Assets/Mirror/Components/NetworkLobbyManager.cs

```diff
@@ -0,0 +1,520 @@
+            {
+                // cant be ready in lobby, add to ready list
+                PendingPlayer pending;
+                pending.conn = conn;
+                pending.lobbyPlayer = lobbyPlayerGameObject;
+                pendingPlayers.Add(pending);
+                return;
+            }
+
+            GameObject gamePlayer = OnLobbyServerCreateGamePlayer(conn);
+            if (gamePlayer == null)
+            {
+                // get start position from base class
+                Transform startPos = GetStartPosition();
+                gamePlayer = startPos != null
+                    ? Instantiate(playerPrefab, startPos.position, startPos.rotation)
+                    : Instantiate(playerPrefab, Vector3.zero, Quaternion.identity);
+                gamePlayer.name = playerPrefab.name;
+            }
+
+            if (!OnLobbyServerSceneLoadedForPlayer(lobbyPlayerGameObject, gamePlayer))
+                return;
+
+            // replace lobby player with game player
+            NetworkServer.ReplacePlayerForConnection(conn, gamePlayer);
+        }
+
+        static int CheckConnectionIsReadyToBegin(NetworkConnection conn)
+        {
+            int countPlayers = 0;
+            if (conn.playerController.gameObject.GetComponent<NetworkLobbyPlayer>().ReadyToBegin)
+                countPlayers += 1;
+
+            return countPlayers;
+        }
+
+        public void CheckReadyToBegin()
+        {
+            if (SceneManager.GetActiveScene().name != LobbyScene) return;
+
+            int readyCount = 0;
+
+            foreach (KeyValuePair<int, NetworkConnection> conn in NetworkServer.connections)
+            {
+                if (conn.Value == null)
+                    continue;
+
+                readyCount += CheckConnectionIsReadyToBegin(conn.Value);
+            }
+
+            if (minPlayers > 0 && readyCount < minPlayers)
+            {
+                allPlayersReady = false;
+                return;
+            }
+
+            pendingPlayers.Clear();
+            allPlayersReady = true;
+            OnLobbyServerPlayersReady();
+        }
+
+        public void ServerReturnToLobby()
+        {
+            if (!NetworkServer.active)
+            {
+                Debug.Log("ServerReturnToLobby called on client");
+                return;
+            }
+            ServerChangeScene(LobbyScene);
+        }
+
+        void CallOnClientEnterLobby()
+        {
+            OnLobbyClientEnter();
+            foreach (NetworkLobbyPlayer player in lobbySlots)
+            {
+                if (player == null)
+                    continue;
+
+                player.OnClientEnterLobby();
+            }
+        }
+
+        void CallOnClientExitLobby()
+        {
+            OnLobbyClientExit();
+            foreach (NetworkLobbyPlayer player in lobbySlots)
+            {
+                if (player == null)
+                    continue;
+
+                player.OnClientExitLobby();
```

Comment:
same here. player != null then onclientexitlobby would be less magic imho

Reply:
Fixed